src/Tutor.API/Controllers/Instructor/Analysis/EventsController.cs

```diff
@@ -2,20 +2,21 @@
 using Dahomey.Json;
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
+using Tutor.BuildingBlocks.Core.Domain.EventSourcing;
 using Tutor.BuildingBlocks.Core.EventSourcing;
+using Tutor.BuildingBlocks.Infrastructure.Database.EventStore.DefaultEventSerializer;
 using Tutor.KnowledgeComponents.Infrastructure.Database.EventStore;
-using Tutor.KnowledgeComponents.Infrastructure.Database.EventStore.DefaultEventSerializer;
 
 namespace Tutor.API.Controllers.Instructor.Analysis;
 
 [Authorize(Policy = "instructorPolicy")]
 [Route("api/analysis/knowledge-components/events")]
-public class EventsController : BaseApiController
+public class EventsController<TEvent> : BaseApiController where TEvent : DomainEvent
```

Comment:
No need for <TEvent> in the controller. I guess it was caused by a lack of focus, but we cannot parametrize all the layers of our application when we only need a paramterized infrastructure service.